h/views/groups.py

```diff
@@ -112,23 +114,12 @@ def _update_group(self, appstruct):
 @view_config(route_name='group_read',
              request_method='GET',
              renderer='h:templates/groups/share.html.jinja2',
-             effective_principals=security.Authenticated)
+             effective_principals=security.Authenticated,
+             custom_predicates=[custom_predicates.has_read_permission,
+                                not_(custom_predicates.feature('search_page'))])
```

Comment:
This is the view function for the classic (pre-activity pages) group page. Tighten up the view config on this so that's it's only called if the user is a member of the group and the search_page feature flag is off.

Note that this single view function used to handle the group URL when the user was or was not a member of the group (as long as they were logged in). This has now been split out so that this view is _only_ for logged in users who _are_ members of the group.